src/core/server/ui_settings/integration_tests/so_migrations.test.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { createTestHarness, SavedObjectTestHarness } from '../../../test_helpers/so_migrations';
+
+/**
+ * These tests are a little unnecessary because these migrations are incredibly simple, however
+ * this file serves as an example of how to use test_helpers/so_migrations.
+ */
+describe('ui settings migrations', () => {
+  let testHarness: SavedObjectTestHarness;
+
+  beforeAll(async () => {
+    testHarness = createTestHarness();
+    await testHarness.start();
+  });
+
+  afterAll(async () => {
+    await testHarness.stop();
+  });
+
+  it('migrates siem:* configs', async () => {
+    const input = [
+      {
+        type: 'config',
+        id: '1',
+        attributes: {
+          'siem:value-one': 1000,
+          'siem:value-two': 'hello',
+        },
+        references: [],
+      },
+    ];
+    expect(await testHarness.migrate(input)).toEqual([
+      expect.objectContaining({
```

Comment:
Core tests shouldn't implicitly depend on the plugin logic. Let the test register a test-specific migration instead.

Reply:
This logic actually lives in Core right now inside `src/core/server/ui_settings/saved_objects/migrations.ts` which is why I used it. I'd like this test to actually exercise the migrations that Core actually contains rather than use a test-specific migration. These are the only ones we have in Core ATM. WDYT?

Reply:
my bad, I forgot we keep it in the Core for time being.

Reply:
No worries 😄